File: problems/1723/solution.go
package main
const MaxInt = int(^uint(0) >> 1)

func minimumTimeRequired(jobs []int, k int) int {
    var backtrack func(int)
    ans, workers := MaxInt, make([]int, k)
    
    sort.Sort(sort.Reverse(sort.IntSlice(jobs)))
    
    backtrack = func (i int) {
        if i == len(jobs) {
            ans = MaxEl(workers)
        } else {
            for j := 0; j < k; j++ {
                if j == 0 || workers[j - 1] > workers[j] {
                    workers[j] += jobs[i]

                    if MaxEl(workers) < ans {
                        backtrack(i + 1)
                    }

                    workers[j] -= jobs[i]
                }
            }
        }
    }
    
    backtrack(0)
    
    return ans
}

func MaxEl(ar []int) int {
    ans := 0
    
    for _, el := range ar {
        if el > ans {
            ans = el
        }
    }
    
    return ans
}

File: problems/1723/solution.cpp
#define ll long long int

class Solution {
public:
    int minimumTimeRequired(vector<int>& jobs, int k) {
        ll ans = INT_MAX;
        vector<ll> workers(k);
        
        sort(jobs.begin(), jobs.end(), greater<ll>());
        
        function<void(int)> bt;
        
        bt = [&](int i) {
            if (i == jobs.size()) {
                ans = *max_element(workers.begin(), workers.end());
            } else {
                for (int j = 0; j < k; j++) {
                    if (j == 0 || workers[j - 1] > workers[j]) {
                        workers[j] += jobs[i];
                        
                        if (*max_element(workers.begin(), workers.end()) < ans) {
                            bt(i + 1);
                        }
                        
                        workers[j] -= jobs[i];
                    }
                }
            }
        };
        
        bt(0);
        
        return ans;
    }
};